maro/rl/policy/trainer.py

```diff
@@ -98,3 +98,50 @@ def trainer_node(
             }
             logger.debug(f"total policy update time: {time.time() - t0}")
             proxy.reply(msg, body=msg_body)
+        elif msg.tag == MsgTag.SCATTER_EXP:
+            t0 = time.time()
+            msg_body = {
+                MsgKey.TRACKER: {}
+            }
+
+            for name, exp in msg.body[MsgKey.EXPERIENCES].items():
+                policy_dict[name].store(exp)
+                msg_body[MsgKey.TRACKER][name] = policy_dict[name].tracker
+
+            logger.debug(f"total scatter experience time: {time.time() - t0}")
+            proxy.reply(msg, body=msg_body)
+        elif msg.tag == MsgTag.GET_GRAD:
+            t0 = time.time()
+            # message: loss of each trainer node
+            msg_body = {
+                MsgKey.GRAD: {},
+                MsgKey.TRACKER: {}
+            }
+
+            for name, _ in msg.body[MsgKey.EXPERIENCES].items():
+                # TODO: add for-loop of train_epochs
+                # for _ in range(policy_dict[name].config.train_epochs):
+                loss = policy_dict[name].get_loss()
+
+                # Collect gradient
+                loss.backward()
+                grad_dict = {}
+                for param_name, param in policy_dict[name].q_net.named_parameters():
```

Comment:
Trainer should be policy agnostic. This code only applies to DQN. We need to think about how to design the interfaces here.